csob/resources/payment/init.py

```diff
@@ -0,0 +1,100 @@
+import json
+from typing import Optional, List
+
+from csob.resources.payment import PaymentCSOBResource
+
+
+class PaymentInit(PaymentCSOBResource):
+    url = 'payment/init'
+    request_signature = (
+        'merchantId', 'orderNo', 'dttm', 'payOperation', 'payMethod', 'totalAmount', 'currency', 'closePayment',
+        'returnUrl', 'returnMethod', 'cart', 'description', 'merchantData', 'customerId', 'language', 'ttlSec',
+        'logoVersion', 'colorSchemeVersion')
+    optional_request_signature = ('merchantData', 'customerId', 'ttlSec', 'logoVersion', 'colorSchemeVersion')
+
+    def post(self, key, order_number: str, pay_operation: str, pay_method: str, total_amount: int, currency: str,
+             close_payment: bool, return_url: str, return_method: str, description: str, language: str,
+             merchant_data: Optional[str], customer_id: Optional[str], cart: Optional[List[dict]] = None,
+             ttl_sec: Optional[int] = None, logo_version: Optional[int] = None,
+             color_scheme_version: Optional[int] = None):
+        local_json = self.get_base_json()
+
+        if len(order_number) > 10:
+            raise ValueError('orderNo is too long.')
+        local_json['orderNo'] = order_number
+
+        if pay_operation not in ['payment', 'oneclickPayment']:
```

Comment:
not in {} btw neměly by to být enumy?

Reply:
no enum tam na to je, ale :

```
>>> 'payment' in PayOperation
False
```

Reply:
tomu nerozumím, můžeš převést klíč enumu na string

Reply:
jakože mám udělat:
```
if pay_operation not in {PayOperation.PAYMENT.value, PayOperation.PAYMENT..value}
```